docs/core/deploying/native-aot/index.md

```diff
@@ -114,24 +122,48 @@ By default, native AOT publishing produces debug information in a separate file
+### [.NET 7](#tab/net7)
+
+Native AOT apps have limitations and compatibility issues:
 
-- No dynamic loading (for example, `Assembly.LoadFile`).
-- No run-time code generation (for example, `System.Reflection.Emit`).
+- No dynamic loading, for example, `Assembly.LoadFile`.
+- No run-time code generation, for example, `System.Reflection.Emit`.
 - No C++/CLI.
-- No built-in COM (only applies to Windows).
+- Windows: No built-in COM.
 - Requires trimming, which has [limitations](../trimming/incompatibilities.md).
 - Implies compilation into a single file, which has known [incompatibilities](../single-file/overview.md#api-incompatibility).
 - Apps include required runtime libraries (just like [self-contained apps](../index.md#publish-self-contained), increasing their size as compared to framework-dependent apps).
 - <xref:System.Linq.Expressions> always use their interpreted form, which is slower than run-time generated compiled code.
+- Should be targeted for console type apps, ASP.NET Core is not supported.
+- Not all the runtime libraries are fully annotated to be native AOT compatible. That is, some warnings in the runtime libraries aren't actionable by end developers.
+- Limited diagnostic support for debugging and profiling.
 
-The publish process analyzes the entire project and its dependencies and produces warnings whenever the limitations of the published application might be encountered at run time.
+The publish process:
 
-The first release of native AOT in .NET 7 has some more limitations:
+- Analyzes the entire project and its dependencies.
+- Produces warnings whenever the limitations of the published app might be encountered at run time.
+
+### [.NET 8+](#tab/net8plus)
 
-- Should be targeted for console type applications (not ASP.NET Core).
-- Not all the runtime libraries are fully annotated to be native AOT compatible (that is, some warnings in the runtime libraries aren't actionable by end developers).
-- Limited diagnostic support (debugging and profiling).
+Native AOT apps have limitations and compatibility issues:
+
+- No dynamic loading, for example, `Assembly.LoadFile`.
+- No run-time code generation, for example, `System.Reflection.Emit`.
+- No C++/CLI.
+- Windows: No built-in COM.
+- Requires trimming, which has [limitations](../trimming/incompatibilities.md).
+- Implies compilation into a single file, which has known [incompatibilities](../single-file/overview.md#api-incompatibility).
+- Apps include required runtime libraries (just like [self-contained apps](../index.md#publish-self-contained), increasing their size as compared to framework-dependent apps).
+- <xref:System.Linq.Expressions> always use their interpreted form, which is slower than run-time generated compiled code.
+- Not all the runtime libraries are fully annotated to be native AOT compatible. That is, some warnings in the runtime libraries aren't actionable by end developers.
+- Limited diagnostic support for debugging and profiling.
```

Comment:
Diagnostic support has been addressed in .NET 8. We have way better diagnostic support in .NET 8. I'm not sure what is left as a limitation.

cc @LakshanF @agocke @tommcdon

Reply:
There are still quite a few limitations and UX differences. Lakshan is working on a doc about the diag story for NativeAOT on .NET 8. Once that is done we should just like to that from here.

Reply:
> There are still quite a few limitations and UX differences. Lakshan is working on a doc about the diag story for NativeAOT on .NET 8. Once that is done we should just ~like~ link to that from here.

#36502

Can we get this approved so I can S&M?